x-test/API-tests/employee_test.go

```diff
@@ -91,6 +92,79 @@ func disableEmployee(postUrl string) error {
 
 }
 
+func TestEmployee_AvoidPhantomIncrements(t *testing.T) {
+	// as the test name suggests this test is to prevent the auto increment in
+	//  the employees table from incrementing without an actual insert. This
+	// test will reveal when a condition exists where an insert causes the
+	// increment to increase but a unique key forces the ON DUPLICATE UPDATE
+	// to update an existing row.
+
+	// add new employee getting the empUID
+	m := Employee{
+		FirstName: "testing",
+		LastName:  "users",
+		UserName:  "testingusers",
+	}
+
+	employeeId, err := postEmployee(NationalOrgchartURL+`api/employee/new`, m)
+
+	if err != nil {
+		t.Error(err)
+	}
+
+	if employeeId == "" {
+		t.Error("no user id returned")
+	}
+
+	var empUID1 string
+
+	empUID1, err = postEmployee(RootOrgchartURL+`api/employee/new`, m)
+
+	if err != nil {
+		t.Error(err)
+	}
+
+	if empUID1 == "" {
+		t.Error("no user id returned")
+	}
+
+	// run refresh Orgchart
+	err = updateEmployees(RootOrgchartURL + `scripts/refreshOrgchartEmployees.php`)
```

Comment:
go-staticcheck: this value of err is never used

Reply:
Added the check.